### ISSUE TITLE
Perplexity nan: bpe with pre_word_vecs_enc and pre_word_vecs_dec

### ISSUE THREAD

**Issue by netxiao** [OpenNMT/OpenNMT#298]
my corpus has been toked with bpe, train config file:

```
start_epoch = 1
max_batch_size = 64
optim = sgd
learning_rate = 1
end_epoch = 13
start_decay_at = 9

min_learning_rate = 0.0001
start_decay_ppl_delta = 0.001
decay =  default

pre_word_vecs_enc = data_it/train_share/vec_src-embeddings-200.t7
pre_word_vecs_dec = data_it/train_share/vec_tgt-embeddings-200.t7

layers = 2
brnn = false
rnn_size = 500

gpuid = 1
fp16 = true
curriculum = 1 
save_every = 5000   
```
**if set pre_word_vecs_enc and pre_word_vecs_dec, ppl keep nan.**

training log:

> [05/30/17 19:06:10 WARNING] The caching CUDA memory allocator is enabled. This allocator improves performance at the cost of a higher GPU memory usage. To optimize for memory, consider disabling it by setting the environment variable: THC_CACHING_ALLOCATOR=0	
> [05/30/17 19:06:10 INFO] Training Sequence to Sequence with Attention model...	
> [05/30/17 19:06:10 INFO] Loading data from './data_it/share-train.t7'...	
> [05/30/17 19:06:39 INFO]  * vocabulary size: source = 23841; target = 35209	
> [05/30/17 19:06:39 INFO]  * additional features: source = 0; target = 0	
> [05/30/17 19:06:39 INFO]  * maximum sequence length: source = 35; target = 36	
> [05/30/17 19:06:39 INFO]  * number of training sentences: 274845	
> [05/30/17 19:06:39 INFO]  * number of batches: 1096	
> [05/30/17 19:06:39 INFO]    - source sequence lengths: equal	
> [05/30/17 19:06:39 INFO]    - maximum size: 256	
> [05/30/17 19:06:39 INFO]    - average size: 250.77	
> [05/30/17 19:06:39 INFO]    - capacity: 100.00%	
> [05/30/17 19:06:39 INFO] Building model...	
> [05/30/17 19:06:39 INFO]  * Encoder:	
> [05/30/17 19:06:40 INFO]    - word embeddings size: 500	
> [05/30/17 19:06:40 INFO]    - type: simple	
> [05/30/17 19:06:40 INFO]    - structure: cell = LSTM; layers = 2; rnn_size = 500; dropout = 0.3	
> [05/30/17 19:06:40 INFO]  * Decoder:	
> [05/30/17 19:06:42 INFO]    - word embeddings size: 500	
> [05/30/17 19:06:42 INFO]    - attention: global (general)	
> [05/30/17 19:06:42 INFO]    - structure: cell = LSTM; layers = 2; rnn_size = 500; dropout = 0.3	
> [05/30/17 19:06:42 INFO]  * Bridge: copy	
> [05/30/17 19:06:46 INFO] Initializing parameters...	
> [05/30/17 19:06:50 INFO]  * number of parameters: 56930709	
> [05/30/17 19:06:50 INFO] Preparing memory optimization...	
> [05/30/17 19:06:50 INFO]  * sharing 69% of output/gradInput tensors memory between clones	
> [05/30/17 19:06:50 INFO] Start training from epoch 1 to 13...	
> [05/30/17 19:06:50 INFO] 	
> [05/30/17 19:07:13 INFO] Epoch 1 ; Iteration 50/1096 ; Optim SGD LR 1.0000 ; Source tokens/s 1647 ; Perplexity nan

**Comment by guillaumekln** [comment]
How did you generate the pretrained words embeddings?

**Comment by netxiao** [comment]
my training script:
`word2vec -train ../train.src.share -output vectors.src.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 2 -binary 1 `

convert to t7 model:
`th tools/embeddings.lua -embed_type word2vec -embed_file data_it/train_share/w2v_en/vectors.src.bin -dict_file ./data_it/dict/17_dict.src.dict  -save_data data_it/train_share/vec_src`

**Comment by guillaumekln** [comment]
Could you try with a previous version of this script?

```bash
cd OpenNMT
luarocks remove opennmt
git checkout fc6f231b905b2a2383f38dc563d35fc3d3b86e3f
th tools/embeddings.lua -embed_type word2vec -embed_file data_it/train_share/w2v_en/vectors.src.bin -dict_file ./data_it/dict/17_dict.src.dict -save_data data_it/train_share/vec_src
git checkout master
```

Then restart the training.

**Comment by jsenellart-systran** [comment]
@netxiao - which version are you using, on version 0.7+, the `embed_type` value should be `word2vec-bin`.

I just reproduced your set-up using toy data and it works end to end:

```
th preprocess.lua -train_src data/src-train.txt -train_tgt data/tgt-train.txt -valid_src data/src-val.txt -valid_tgt data/tgt-val.txt -save_data data/demo -src_vocab_size 20000 -tgt_vocab_size 20000
word2vec -train data/src-train.txt -output data/src-vec.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 2 -binary 1
word2vec -train data/tgt-train.txt -output data/tgt-vec.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 2 -binary 1
th tools/embeddings.lua -embed_type word2vec-bin -embed_file data/src-vec.bin -dict_file data/demo.src.dict  -save_data data/src-vec_demo
th tools/embeddings.lua -embed_type word2vec-bin -embed_file data/tgt-vec.bin -dict_file data/demo.tgt.dict  -save_data data/tgt-vec_demo
th train.lua -data data/demo-train.t7 -save_model model -rnn_size 200 -word_vec_size 200 -pre_word_vecs_enc data/src-vec_demo-embeddings-200.t7 -pre_word_vecs_dec data/tgt-vec_demo-embeddings-200.t7
[...]
[05/31/17 16:11:20 INFO] Epoch 1 ; Iteration 50/184 ; Optim SGD LR 1.0000 ; Source tokens/s 653 ; Perplexity 21212.43
```

**Comment by netxiao** [comment]
thanks! it works for me.
